packages/gateway/src/gateway.js

```diff
@@ -279,6 +279,12 @@ async function updateGatewayRedirectCounter(request, env) {
  * @param {string} gwUrl
  */
 async function getGatewayRateLimitState(request, env, gwUrl) {
+  if (!shouldCheckRateLimitPrevention(gwUrl)) {
```

Comment:
We can make DO object being one per gateway here, Keeping the noop requests to ipfs.io and cf-ipfs. I don't think that it should be needed and we should really look into decommissioning this DO